docs/en/getting-started/get-started-stack.asciidoc

```diff
@@ -0,0 +1,871 @@
+----------------------------------------------------------------------
+sudo /etc/init.d/logstash start
+----------------------------------------------------------------------
+
+*rpm:*
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+sudo service logstash start
+----------------------------------------------------------------------
+
+*mac:*
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+./bin/logstash -f logstash.conf
+----------------------------------------------------------------------
+
+*win:*
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+bin\logstash.bat -f logstash.conf
+----------------------------------------------------------------------
+
+NOTE: The default configuration for {beats} and {ls} uses plain TCP. For
+encryption you must explicitly enable SSL when you configure {beats} and {ls}.
+
+{ls} reads from the {beats} input and indexes events into {es}. You haven't
+defined a filter section yet, so {ls} simply forwards events to {es} without
+additional processing. Next, you learn how to define the filter stage.
+
+[float]
+[[logstash-filter]]
+==== Define a filter to extract data from a field
+
+The system metrics collected by {metricbeat} include a field called `cmdline`
+that contains the full command-line argument used to start the process. For
+example:
+
+[source,json]
+----
+"cmdline": "/Applications/Firefox.app/Contents/MacOS/plugin-container.app/Contents/MacOS/plugin-container -childID 3
+-isForBrowser -boolPrefs 36:1|299:0| -stringPrefs 285:38;{b77ae304-9f53-a248-8bd4-a243dbf2cab1}| -schedulerPrefs
+0001,2 -greomni /Applications/Firefox.app/Contents/Resources/omni.ja -appomni
+/Applications/Firefox.app/Contents/Resources/browser/omni.ja -appdir
+/Applications/Firefox.app/Contents/Resources/browser -profile
+/Users/dedemorton/Library/Application Support/Firefox/Profiles/mftvzeod.default-1468353066634
+99468 gecko-crash-server-pipe.99468 org.mozilla.machname.1911848630 tab"
+----
+
+
+You might decide that you don't want to send the whole command-line argument to
+{es}. You can write a grok filter that matches the first part of the command
+and saves it to a new field. You can also remove the original field, `cmdline`,
+from the event. For example:
+
+// REVIEWERS: I took a stab at this, but I'm not very good at grok or
+// regex. In fact, I sort of hate it...a lot. Can someone review and suggest
+// something better based on the format you would expect in this field?
+
+[source,ruby]
+----
+filter {
+  if [system][process] {
+    if [system][process][cmdline] {
+      grok {
+        match => { "[system][process][cmdline]" => ["%{CMD:[system][process][short_cmdline]}"] }
+        pattern_definitions => {
+          "CMD" => "(?:/[A-Za-z0-9$.+!*'(){},~:;=@#%&_\-]*)+\s*-*[A-Za-z0-9$.+!*'(){},~:;=@#%&_\-]*"
```

Comment:
This pattern is one heck of an introduction to GROK and regular expressions; can we start with something simpler?

What about capturing the just the cmd's path, which would allow us to use a pre-defined `PATH` pattern? That would just be:

~~~
filter {
  if [system][process] {
    if [system][process][cmdline] {
      grok {
        match => {
          "[system][process][cmdline]" => "^%{PATH:[system][process][cmdline_path]}"
        }
        remove_field => "[system][process][cmdline]"
      }
    }
  }
}
~~~

Reply:
Make sense! Ironically, that's where I started, but then went down a rabbit hole...I thought I wanted to capture the first argument, but backed away from that because it got weird quickly.